btree/btree.c

```diff
@@ -1772,6 +1762,29 @@ static void tree_put(struct td *tree)
 
 static const struct node_type fixed_format;
 
+static int64_t seg_node_alloc(struct node_op *op, struct td *tree, int shift,
+			      const struct node_type *nt, struct m0_be_tx *tx,
+			      int nxt)
+{
+	void          *area;
+	int            size = 1ULL << shift;
+
+	M0_PRE(op->no_opc == NOP_ALLOC);
+	M0_PRE(node_shift_is_valid(shift));
+	area = m0_alloc_aligned(size, shift);
+	M0_ASSERT(area != NULL);
+	op->no_addr = segaddr_build(area, shift);
+	op->no_tree = tree;
+	return nxt;
+}
+
+static void seg_node_free(struct node_op *op, int shift, struct m0_be_tx *tx,
+			     int nxt)
+{
+	m0_free_aligned(segaddr_addr(&op->no_addr), 1ULL << shift, shift);
+	/** Capture in transaction */
+}
+
```

Comment:
It seems you have just changed the names of the functions mem_node_*() to seg_node_*() !!!
How will that help !!!

Reply:
Updated the code.